### ISSUE TITLE
Popup closeOnClick working backwards?

### ISSUE THREAD

**Issue by disneyquesting** [visgl/react-map-gl#1389]
Hey everybody. Not sure if i'm just confused or not but when trying to use react-map-gl popup with markers, I noticed one of the props you can send to the popup is this:'

closeOnClick (Boolean)

    default: true

If true, the popup will closed when the map is clicked.

However, upon implementation the result is backwards; the popup will only close if you click within the popup and not on the map itself.

**Comment by Pessimistress** [comment]
I think the documentation is incorrect. The prop is intended to close the popup when the pointer clicks anywhere inside the popup.

**Comment by disneyquesting** [comment]
Thank you, that makes sense.

**Comment by hi019** [comment]
> I think the documentation is incorrect. The prop is intended to close the popup when the pointer clicks anywhere inside the popup.

The Mapbox [Popup docs](https://docs.mapbox.com/mapbox-gl-js/api/markers/) say this option will close the popup when the map is clicked. So I think this issue is valid

**Comment by FullStackJohnMack** [comment]
Bump on this

**Comment by iserranoe** [comment]
I'm having the same issue. The option says "the popup will closed when the map is clicked", so it should close when clicking on the map outside the popup, otherwise, the documention is incorrect.

**Comment by Pessimistress** [comment]
Thanks for reporting the discrepency. This is fixed in v5.3.17/v6.1.17.

**Comment by easbar** [comment]
Thanks for fixing this! I confirm it is working. I tried it using your "controls" example (just setting `closeOnClick` to `true`). My use-case is slightly different: I would like to open the popup on right-click. So I tried it like this (also in the example/controls/src/app.js):

```typescript
   <MapGL
   // ... a few lines omitted
  onContextMenu={e => {
          e.preventDefault()
          setPopupInfo({ longitude: e.lngLat[0], latitude: e.lngLat[1] })
  }}
  >
```

The popup opens, but I need to hold the right mouse button for a moment for it to stay open. Otherwise it closes immediately. So a single right-click opens the popup and closes it again immediately. If I manage to open it the behavior is as expected though: Clicking the map somewhere closes it.

I confirmed that until 5.3.16 the behavior was different: Right-clicking the map opened the popup as expected, but clicking the map afterwards it did not close. So it seems like this was fixed in #1565, but now there is another issue that I described above.
